### ISSUE TITLE
Multi-class sample imbalance problem

### ISSUE THREAD

**Issue by lyliu** [microsoft/LightGBM#3167]
Hi, 
I have sample imbalance problem for multiclass.
I use class_weight, but it doesn't  have any effect on the results, I think it may not work, so how to solve the multi-class sample imbalance problem.

**Comment by guolinke** [comment]
can you provide a code example and demonstrate how it doesn't work?

**Comment by lyliu** [comment]
My code example is as follows:
`data_file='data/train_data/train_data.csv'
train_data,test_data = get_data(data_file)
class_weight = dict({1:0.1, 2:0.3, 3:0.3, 4:0.3})

print("=============Begin training model=============")
# set param
param = {'num_leaves':100,
        'min_data_in_leaf':30, 
        'objective':'multiclass',
        'max_depth':10,
        'learning_rate':0.1,
        'min_child_samples':20,
        "boosting":"gbdt",
        "feature_fraction":0.9,
         "bagging_freq":1,
         "bagging_fraction":0.9,
         "bagging_seed":11,
        "metric":'multiclass',
         "lambda_l1":0.1,
         "verbosity":-1,
         "nthread":4,
         "random_state":4590,
        "num_class":5,
         "class_weight":class_weight
        }

# training
# Training a model requires a parameter list and data set:

num_round = 500
bst = lgb.train(param, train_data, num_round, valid_sets=[test_data], early_stopping_rounds=10)
bst.save_model('model.txt', num_iteration=bst.best_iteration)

`
I set "class_weight" in "param", but the accuracy, recall and accuracy have not changed, so I think  it doesn't work.

**Comment by StrikerRUS** [comment]
@lyliu `class_weight` is defined only in [sklearn wrapper](https://lightgbm.readthedocs.io/en/latest/Python-API.html#scikit-learn-api). It cannot be used in ordinary API.
